File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int , int> test;
        for(int &i : nums){
            test[i]++;
            if(test[i] > 1){
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        dp= [-1]*n

        def rob(i):
            if i >= n: # We are out of houses
                return 0
            if dp[i]!=-1:
                return dp[i]
            
            dp[i] = max(nums[i]+ rob(i+2) , rob(i+1))
            return dp[i]
        
        return rob(0)